src/test/java/codeu/controller/LoginServletTest.java

```diff
@@ -80,26 +70,35 @@ public void testDoPost_NewUser() throws IOException, ServletException {
 
     Mockito.verify(mockUserStore).addUser(userArgumentCaptor.capture());
     Assert.assertEquals(userArgumentCaptor.getValue().getName(), "test username");
-
+    Assert.assertEquals(userArgumentCaptor.getValue().getPassword(), "test password");
+    
     Mockito.verify(mockSession).setAttribute("user", "test username");
+    Mockito.verify(mockSession).setAttribute("password", "test password");
+    
     Mockito.verify(mockResponse).sendRedirect("/conversations");
   }
 
   @Test
   public void testDoPost_ExistingUser() throws IOException, ServletException {
     Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
-
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("test password");
+    
     UserStore mockUserStore = Mockito.mock(UserStore.class);
     Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(true);
     loginServlet.setUserStore(mockUserStore);
-
+    
     HttpSession mockSession = Mockito.mock(HttpSession.class);
     Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
-
+    
+    User mockUser = Mockito.mock(User.class);
+    Mockito.when(mockUser.getName()).thenReturn("test username");
+    Mockito.when(mockUser.getPassword()).thenReturn("test password");
+    Mockito.verify(mockUserStore).addUser(mockUser);
+    
     loginServlet.doPost(mockRequest, mockResponse);
 
     Mockito.verify(mockUserStore, Mockito.never()).addUser(Mockito.any(User.class));
-
+   
     Mockito.verify(mockSession).setAttribute("user", "test username");
     Mockito.verify(mockResponse).sendRedirect("/conversations");
   }
```

Comment:
You're missing tests for the new login use cases.  Specifically:

- User doesn't exist
- User password is wrong.